### PR TITLE
tests: Upgrade Travis OS to Xenial

### PR DESCRIPTION
**Pull request by gkrizek**
Update base Travis OS to `xenial` from `trusty`. 

Link to Travis Docs for Xenial: https://docs.travis-ci.com/user/reference/xenial/

As noted in the documentation, Docker version is also updated from `17.06` to `18.06`

Also includes:
    - If running Bitcoin config with LSan sanitizer, Allow ptrace in Docker run command
    - Increase timeout of feature_assumevalid test to fix flaky tests

### CODE REVIEW

**Comment by gkrizek** [comment]
This also updates the compilers and build toolchain, but that shouldn't matter on the host because everything is ran inside of Docker, correct?

```
clang and llvm 7
cmake 3.12.4
gcc 5.4.0
ccache 3.2.4
```
https://docs.travis-ci.com/user/reference/xenial/#compilers-and-build-toolchain

**Comment by gkrizek** [comment]
I can't find a legitimate reason by looking through the Docker changelogs or documentation, but when running tests with `LSan` sanitizer, Docker requires the `--cap-add SYS_PTRACE` flag set. 

See relevant issue here:
https://github.com/google/sanitizers/issues/764

**Comment by MeshCollider** [comment]
@gkrizek please stop opening and closing this :)

**Comment by gkrizek** [comment]
Sorry, @MeshCollider. The first time was an accident then the second was to trigger a new CI run. One of the tests seems flaky, but I’ll test that in my fork. :)

**Comment by Empact** [comment]
@gkrizek To trigger a new build, you can set up an account at travis-ci.org and control the builds under: http://travis-ci.org/gkrizek/bitcoin

**Comment by gkrizek** [comment]
Thanks @Empact. I was doing that earlier and it was passing in my fork. That's why I thought this was good to go, but now a test is failing so I need to investigate that more.

**Comment by gkrizek** [comment]
This is complete and ready for review. I've fixed some issues with tests and builds are passing consistently.

File: .travis/test_04_install.sh [hunk 1]
```diff
@@ -15,6 +15,8 @@ export UBSAN_OPTIONS="suppressions=${TRAVIS_BUILD_DIR}/test/sanitizer_suppressio
 env | grep -E '^(BITCOIN_CONFIG|CCACHE_|WINEDEBUG|LC_ALL|BOOST_TEST_RANDOM|CONFIG_SHELL|(ASAN|LSAN|TSAN|UBSAN)_OPTIONS)' | tee /tmp/env
 if [[ $HOST = *-mingw32 ]]; then
   DOCKER_ADMIN="--cap-add SYS_ADMIN"
+elif [[ $BITCOIN_CONFIG = *--with-sanitizers=*address* ]]; then # If ran with (ASan + LSan), Docker needs access to ptrace
```
Comment (Empact) [inline .travis/test_04_install.sh]: nit: Maybe worth linking to one of these?
https://github.com/google/sanitizers/issues/916
https://github.com/google/sanitizers/issues/764

**Comment by fanquake** [comment]
This should probably be broken into two or three commits. i.e the Travis change (with/without the sanitizer change) and then the test timeout change in a separate commit.

**Comment by MarcoFalke** [comment]
utACK b6f0db69a9c9cdf101371720351935121590d3aa. We don't use any features from the travis image other than docker, so changing it shouldn't matter.

**Comment by fanquake** [comment]
Thanks @MarcoFalke. If anything we might get [slightly faster build times](https://docs.travis-ci.com/user/reference/xenial/#differences-from-the-trusty-images).

utACK b6f0db6 - appveyor is the failure.

**Comment by gkrizek** [comment]
It looks like this ran about 5 minutes faster than another PR I have open. So I haven’t done testing to compare speeds, but it does seem faster.

**Comment by MarcoFalke** [comment]
Oh nice. Thanks for the speedup

**Pull request merged**